metecho/api/serializers.py

```diff
@@ -321,6 +322,15 @@ class ReviewSerializer(serializers.Serializer):
     )
 
 
+class CircumspectJSONField(JSONField):
+    def to_representation(self, value):
+        user = getattr(self.context, "request", {}).get("user", None)
+        owner = getattr(self.parent.instance, "owner", None)
+        if owner and user != owner:
+            value = {}
+        return super().to_representation(value)
```

Comment:
@wlonk This doesn't seem to work as expected, and doesn't match how we're accessing the `request` object elsewhere. Is there a better/more-consistent way to do this?

- `user = getattr(self.context.get("request"), "user", None)` works, which is what we're doing elsewhere
- apparently `self.parent.instance` is sometimes a `SoftDeleteQuerySet`, and other times it's a `ScratchOrg` -- won't this only work when it's a model?

Reply:
@duggiemitchell This is the bug you were seeing, by the way.

Reply:
For some reason when I was testing it, I was getting a `request` attribute, not a `"request"` key. I can't find what I was doing, but yeah, the change you suggest seems like the needed one.

Regarding `self.parent.instance`, you're right; I forgot about the `many=True` case, as I repeatedly do. I can fix both.

Reply:
Yeah, sometimes it looks like it's a `Request(user=<channels.auth.UserLazyObject object...>)`, and other times a `<rest_framework.request.Request object...>` -- but in both cases under `{'request': ...}`.

I just really don't understand how this all interacts. 🤷

Reply:
> For some reason when I was testing it, I was getting a `request` attribute, not a `"request"` key. I can't find what I was doing, but yeah, the change you suggest seems like the needed one.
> 
> Regarding `self.parent.instance`, you're right; I forgot about the `many=True` case, as I repeatedly do. I can fix both.

@wlonk Do you have a moment to fix these?